### PR TITLE
feat: add Linux Volume Manager input plugin

### PR DESCRIPTION
**Pull request by powersj**
The Logical Volume Management (LVM) input plugin collects information about
physical volumes, volume groups, and logical volumes. This will only
work on Linux and *BSDs as OS X and Windows do not have support for
these types of disks.

The metrics are broken out into the various LVM types.

Resolves: #345

### Required for all PRs:

- [x] Updated associated README.md.
- [x] Wrote appropriate unit tests.
- [x] Pull request title or commits are in [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/#summary) (e.g. feat: or fix:)

### CODE REVIEW

File: plugins/inputs/lvm/lvm.go [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+	execCommand = exec.Command
+)
+
+var sampleConfig = `
+## Use sudo to run LVM commands
+use_sudo = false
+`
+
+type LVM struct {
+	UseSudo bool
```
Comment (sspaink) [inline plugins/inputs/lvm/lvm.go:26]:
```suggestion
	UseSudo bool `toml:"use_sudo"`
```

File: plugins/inputs/lvm/lvm.go [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+		return err
+	}
+
+	var report pvsReport
+	err = json.Unmarshal(out, &report)
+	if err != nil {
+		return fmt.Errorf("failed to unmarshal physical volume JSON: %s", err)
+	}
+
+	for _, pv := range report.Report[0].Pv {
```
Comment (sspaink) [inline plugins/inputs/lvm/lvm.go:70]: Should check the length of `report.Report[0]` before trying to access the first element. Don't forget to update the same loops below.

**Review by sspaink — CHANGES_REQUESTED** [review]
Thanks for working on this, some minor comments.

File: plugins/inputs/lvm/lvm.go (hunk 1 above)
Comment (powersj) [reply plugins/inputs/lvm/lvm.go]: Good catch!

File: plugins/inputs/lvm/lvm.go (hunk 2 above)
Comment (powersj) [reply plugins/inputs/lvm/lvm.go]: Ah right thank you!

**Comment by powersj** [comment]
Comments addressed, read for re-review

**Review by sspaink — APPROVED** [review]
lgtm!

**Comment by rafaelreis-r** [comment]
Hey @powersj !

Just tried using your pluigin on Ubuntu 18.04 and noticed the pvs bin path is hardcoded:

`[inputs.lvm] Error in plugin: failed to run command sudo -n /usr/sbin/pvs --reportformat json --units b --nosuffix -o pv_name,vg_name,pv_size,pv_free,pv_used: exit status 1 -`

` which pvs
/sbin/pvs`

Could you maybe expand the configuration to allow different paths or use system envs to do so?

**Comment by c4baceira** [comment]
hello @powersj ! 

Is this plugin ready to be used by the docker version of Telegraf?

I'm getting: 

`2022-06-15T15:22:30Z E! [inputs.lvm] Error in plugin: failed to run command /usr/sbin/pvs --reportformat json --units b --nosuffix -o pv_name,vg_name,pv_size,pv_free,pv_used: fork/exec /usr/sbin/pvs: no such file or directory - `

Using:

 ```
version: '3'

services:

  telegraf:
    image: telegraf
    container_name: telegraf
    restart: always
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock:z
      - ./telegraf/telegraf.conf:/etc/telegraf/telegraf.conf:ro
    links:
      - influxdb
    ports:
      - "8092:8092/udp"
      - "8094:8094"
      - "8125:8125/udp"
    networks:
      - tick
    user: telegraf:118
```

```
telegraf --version
Telegraf 1.23.0 (git: HEAD 806dc283)
```

**Comment by powersj** [comment]
@c4baceira,

The docker image does not contain all the possible dependencies that every plugins would require. If we did, the docker image size would be huge to meet those requirements.

If you want to use this plugin with docer you will need to create a custom image that includes those binaries as well.

**Pull request merged**